crates/nix_rs/src/command.rs

```diff
@@ -174,6 +174,31 @@ impl NixCmd {
         }
     }
 
+    /// Like [Self::run_with_returning_stdout] but inherits stderr from parent to trace the progress
+    /// of running Nix with [Command] customizations.
+    pub async fn trace_run_with_returning_stdout<F>(&self, f: F) -> Result<Vec<u8>, CommandError>
```

Comment:
Can we define this using `run_with` (refactoring the function along the way)?

I wrote `run_with` so that we don't have to create new functions like this.

Reply:
changed `run_with` to remove `trace_run_with_returning_stdout`: https://github.com/juspay/omnix/pull/298/commits/5b83c36ed845d32d5a7d9f897b748cdbd5522d0a